src/components/ha-camera-stream.ts

```diff
@@ -0,0 +1,211 @@
+import {
+  property,
+  PropertyValues,
+  LitElement,
+  TemplateResult,
+  html,
+  CSSResult,
+  css,
+  customElement,
+} from "lit-element";
+
+import computeStateName from "../common/entity/compute_state_name";
+import { HomeAssistant, CameraEntity } from "../types";
+import { fireEvent } from "../common/dom/fire_event";
+import {
+  CAMERA_SUPPORT_STREAM,
+  fetchStreamUrl,
+  computeMJPEGStreamUrl,
+} from "../data/camera";
+import { supportsFeature } from "../common/entity/supports-feature";
+
+type HLSModule = typeof import("hls.js");
+
+@customElement("ha-camera-stream")
+class HaCameraStream extends LitElement {
+  @property() public hass?: HomeAssistant;
+  @property() public stateObj?: CameraEntity;
+  @property({ type: Boolean }) public showControls = false;
+  @property() private _attached = false;
+  // We keep track if we should force MJPEG with a string
+  // that way it automatically resets if we change entity.
+  @property() private _forceMJPEG: string | undefined = undefined;
+  private _hlsPolyfillInstance?: Hls;
+
+  public connectedCallback() {
+    super.connectedCallback();
+    this._attached = true;
+  }
+
+  public disconnectedCallback() {
+    super.disconnectedCallback();
+    this._attached = false;
+  }
+
+  protected render(): TemplateResult | void {
+    if (!this.stateObj || !this._attached) {
+      return html``;
+    }
+
+    return html`
+      ${this._shouldRenderMJPEG
+        ? html`
+            <img
+              @load=${this._elementResized}
+              .src=${__DEMO__
+                ? "/demo/webcamp.jpg"
+                : computeMJPEGStreamUrl(this.stateObj)}
+              .alt=${computeStateName(this.stateObj)}
+            />
+          `
+        : html`
+            <video
+              autoplay
+              muted
+              playsinline
+              ?controls=${this.showControls}
+              @loadeddata=${this._elementResized}
+            ></video>
+          `}
+    `;
+  }
+
+  protected updated(changedProps: PropertyValues) {
+    super.updated(changedProps);
+
+    const stateObjChanged = changedProps.has("_attached");
```

Comment:
This is the same as below?  Should it be different?

Reply:
oh no, that should be `stateObj`